src/transformers/models/bark/processing_bark.py

```diff
@@ -0,0 +1,295 @@
+        voice_preset=None,
+        return_tensors="pt",
+        max_length=256,
+        add_special_tokens=False,
+        return_attention_mask=True,
+        return_token_type_ids=False,
+        **kwargs,
+    ):
+        """
+        Main method to prepare for the model one or several sequences(s). This method forwards the `text` and `kwargs`
+        arguments to the AutoTokenizer's [`~AutoTokenizer.__call__`] to encode the text. The method also proposes a
+        voice preset which is a dictionary of arrays that conditions `Bark`'s output. `kwargs` arguments are forwarded
+        to the tokenizer and to `cached_file` method if `voice_preset` is a valid filename.
+
+        Args:
+            text (`str`, `List[str]`, `List[List[str]]`):
+                The sequence or batch of sequences to be encoded. Each sequence can be a string or a list of strings
+                (pretokenized string). If the sequences are provided as list of strings (pretokenized), you must set
+                `is_split_into_words=True` (to lift the ambiguity with a batch of sequences).
+            voice_preset (`str`, `Dict[np.ndarray]`):
+                The voice preset, i.e the speaker embeddings. It can either be a valid voice_preset name, e.g
+                `"en_speaker_1"`, or a directly a dictionnary of embeddings for each submodel of `Bark`. Or it can be a
+                valid file name of a local `.npz` single voice preset.
+            return_tensors (`str` or [`~utils.TensorType`], *optional*):
+                If set, will return tensors of a particular framework. Acceptable values are:
+
+                - `'pt'`: Return PyTorch `torch.Tensor` objects.
+                - `'np'`: Return NumPy `np.ndarray` objects.
+
+        Returns:
+            Tuple([`BatchEncoding`], Dict[~utils.TensorType]): A tuple composed of a [`BatchEncoding`], i.e the output
+            of the `tokenizer` and a `Dict[~utils.TensorType`]`, i.e the voice preset with the right tensors type.
+        """
+        if voice_preset is None or isinstance(voice_preset, dict):
+            pass
+        else:
+            voice_preset_key = voice_preset.replace("/", "_")
+            if isinstance(voice_preset, str) and f"{voice_preset_key}_semantic_prompt" in self.speaker_embeddings_dict:
+                voice_preset = {}
+                for prompt_type in [
+                    "semantic_prompt",
+                    "coarse_prompt",
+                    "fine_prompt",
+                ]:
+                    voice_preset[prompt_type] = self.speaker_embeddings_dict[f"{voice_preset_key}_{prompt_type}"]
+
+            else:
+                if isinstance(voice_preset, str) and not voice_preset.endswith(".npz"):
+                    voice_preset = voice_preset + ".npz"
+
+                voice_preset = np.load(voice_preset)
+
+        self._validate_voice_preset_dict(voice_preset)
+        voice_preset = convert_dict_to_tensors(voice_preset, return_tensors)
```

Comment:
Might it be easier to just convert each item of the dict on the fly rather than defining a new function `convert_dict_to_tensors`? Would be in favour of minimising the amount of extra code required for functionality that already exists

```python
for prompt_type in voice_preset:
    voice_preset[prompt_type] = voice_preset[prompt_type].convert_to_tensors(return_tensors)
```

Reply:
Actually what you might have to do is first convert `voice_preset` to a `BatchFeature` with `self.feature_extractor.pad` and then call `.convert_to_tensors`

Reply:
I didn't know `BatchFeature` was a thing, it's much more convenient!